### ISSUE TITLE
Keccak256Hash 일치여부 검증 오류

### ISSUE THREAD

**Issue by Gi-Woong** [UXMUnderGraduate/22-S-Backend#87]
**버그 설명**

음원업로드시 입력한 정보로 만든 keccak256Hash가 실제 컨트렉트상의 keccak256Hash값과 일치함에도 
백엔드에서 올바르지 않은 컨트렉트라고 감지함. 

---

**동작방식**
동작을 재현하는 단계:
22-S-Front에서 PR (#105)이 merge되었다고 가정했을 때,
1. merge된 버젼의 repo에서 `npm start` 로 react 실행
2. producer 계정으로 로그인 또는 회원가입 후 음원 업로드 절차 진행
3. F12로 업로드시 console log 확인
4. (Bad Request) error 발생 확인

---

**스크린샷**
![image](https://user-images.githubusercontent.com/77622897/204086360-6be24a23-4fd5-4996-ba82-c7e6a9f5bfbd.png)
![image](https://user-images.githubusercontent.com/77622897/204086369-75725db6-23cd-4ff3-bc69-33541ebd21ae.png)

---

**추가 컨텍스트**
따로 테스트 코드를 작성하여 
uploadPage를 통해 생성한 contract의 keccak256Hash와 곡 입력에 사용된 정보로 생성한 keccak256Hash가 일치함을 확인함.
![image](https://user-images.githubusercontent.com/77622897/204086284-58ef66b7-6fbf-48e3-8462-39bc4720c493.png)

- 원인을 알고자 backEnd repo를 다운받아 로컬로 실행시켜보려 했지만 에러가 발생하여 원인을 알 수 없었고, 
- 따로 keccak256Hash 생성 코드를 복사 후 자체 테스트코드에 적용하여 실행시켜보았지만 두 해시가 일치하는 것을 확인하였음.
- backEnd의 infura api를 사용해서도 같은 결과(두 해시가 일치함)가 나왔음.

**Comment by SeonjaeHyeon** [comment]
백엔드 로컬 서버에서 어떤 에러가 발생하는지 정확히 알려주시면 좋을 것 같습니다.  
`keccak256` 해시 검증 오류는 한번 확인해보겠습니다!

**Comment by SeonjaeHyeon** [comment]
로컬 환경에서 정상적으로 로그인되는 것 확인했습니다.  
![화면 캡처 2022-11-26 212001](https://user-images.githubusercontent.com/68897720/204088678-7181851f-9986-4d26-89ad-2d5cda26c230.png)
![화면 캡처 2022-11-26 212100](https://user-images.githubusercontent.com/68897720/204088681-3bd79de2-921e-4969-977d-0c08e88e14b2.png)

**Comment by SeonjaeHyeon** [comment]
@Gi-Woong 
`keccak256` 해시가 동일한 정보로 나오는것 확인했고, 음원 업로드도 정상적으로 완료되는 것 확인했습니다.   
진짜 백엔드 서버 측 버그가 맞는지 다시 한번 확인 부탁드리겠습니다!  

![화면 캡처 2022-11-26 213737](https://user-images.githubusercontent.com/68897720/204089436-6b6a938f-e5f5-45ef-bc06-953151487bf4.png)
![화면 캡처 2022-11-26 213757](https://user-images.githubusercontent.com/68897720/204089437-2150f850-6e24-4004-9e86-aeaffc7af3ae.png)
![화면 캡처 2022-11-26 213825](https://user-images.githubusercontent.com/68897720/204089439-f69838b9-37b9-42d8-baed-cc0ec9d8b002.png)
![화면 캡처 2022-11-26 213839](https://user-images.githubusercontent.com/68897720/204089440-7c2e1d0a-5585-483a-ae89-b2dc2f7d5775.png)

**Comment by Gi-Woong** [comment]
@SeonjaeHyeon  Front쪽은 지금 PR (22-S-Front #105) 올린 버젼으로 사용하신건가요?

**Comment by SeonjaeHyeon** [comment]
네 pull request 올리신 [main](https://github.com/Gi-Woong/22-S-Front) 브랜치 클론해서 테스트하였습니다.

**Comment by Gi-Woong** [comment]
@SeonjaeHyeon 그리고 로컬서버 말고 실제 동작 서버와 연결된 프론트엔드에서 로그인 후  업로드 절차를 정식(곡 저작권자 id 등록, 정산 비율 모두 입력)으로 거쳐 업로드시 로그가 어떻게 나오는지 확인 부탁드립니다!

**Comment by SeonjaeHyeon** [comment]
실제 Oracle 서버에서 테스트한 결과입니다. 업로드 기능은 정상적으로 작동하고 있습니다.  
![화면 캡처 2022-11-26 221442](https://user-images.githubusercontent.com/68897720/204090921-fee8a994-30d4-4700-a309-071f02744909.png)
![화면 캡처 2022-11-26 221504](https://user-images.githubusercontent.com/68897720/204090925-d2bad2f9-0d31-4307-9e14-29c5cb3d50f2.png)

**Comment by Gi-Woong** [comment]
확인 감사합니다.

**Comment by Gi-Woong** [comment]
업로드 페이지에 저작권자 목록 3명 입력시 모두 동일 계정으로 저작권자를 입력하여 등록을 시도할 경우 업로드 에러가 발생하는 것이 확인되었습니다. 
원인은 차후 살펴봐야 할 것 같고 일단 업로드시 3명 다 다른 계정으로 선택하여 업로드할 경우 업로드가 에러 없이 성공하는 것이 확인되었습니다.
